src/page.ts

```diff
@@ -114,11 +123,38 @@ export class Page extends EventTarget {
       );
     });
 
+    if (options?.sandbox) {
+      this.#celestial.addEventListener("Fetch.requestPaused", async (e) => {
+        const { requestId } = e.detail;
+        if (!await this.#validateRequest(e.detail)) {
+          return this.#celestial.Fetch.failRequest({
+            requestId,
+            errorReason: "AccessDenied",
+          });
+        }
+        return this.#celestial.Fetch.continueRequest({ requestId });
+      });
+    }
+
     this.mouse = new Mouse(this.#celestial);
     this.keyboard = new Keyboard(this.#celestial);
     this.touchscreen = new Touchscreen(this.#celestial);
   }
 
+  //TODO(@lowlighter): change "query" by "request" https://github.com/denoland/deno/issues/14668
+  async #validateRequest({ request }: Fetch_requestPausedEvent["detail"]) {
+    const { protocol, host, pathname: path } = new URL(request.url);
+    if (host) {
+      const { state } = await Deno.permissions.query({ name: "net", host });
+      return (state === "granted");
+    }
+    if (protocol === "file:") {
+      const { state } = await Deno.permissions.query({ name: "read", path });
+      return (state === "granted");
+    }
+    return true;
```

Comment:
Does this make sense? When does this condition trigger?

Reply:
I don't think it's possible to reach the `return true` statement, but in the case that it could exist requests that are not using `file:` protocol and don't have a host then they'd be systematically rejected if nothing was returned.
AFAIK it doesn't exist, but maybe there are some edge cases.

As a side note, I did try using `data:` scheme and it totally bypass the sandbox as it does not trigger the `Fetch` namespace at all when passed

Reply:
Good to know, thanks for investigating. We can change later if we realize that this is not correct.